Added a lock function to burgs

Legogizmo opened the pull request:
This adds a burg lock feature which can be accessed from the "burgs overview" and "burg editor" windows seen below.
![Walkthrough1](https://user-images.githubusercontent.com/1015483/112576783-f646bb80-8dc0-11eb-8af2-a8ea93f69755.png)
![Walkthrough3](https://user-images.githubusercontent.com/1015483/112576790-f9da4280-8dc0-11eb-888b-3e88ce712836.png)

The primary use is to keep selected burgs when using the regenerate Burgs feature. 
Other uses include:

- Locked burgs will not be renamed by the "regenerate burg names" in the burgs overview window
- Will not be deleted when using the remove all burgs option in the burgs overview window
- Will not have their population changed by regenerate population tool

Locked burgs can still be modified in the following ways:

- Will still be deleted if the group they are assigned to is removed
- Groups will still be re-assigned by regenerating burgs or states
- Capital status may still be changed by regenerating states
- Culture can still be changed
- The seed for Medieval Fantasy City Generator will be changed unless a custom one is set
- Emblems will sometimes change or stay the same depending on the order in which you change the culture and state, Regenerating Emblems will always change the emblem.

Azgaar commented:
Thanks, looks good, I will try to review it today.

Azgaar commented:
Hello. It looks really good! Just a few minor points I would like to amend:
* Burg name should stay as is when Cultures names are regenerated (via Cultures Editor)
* Lock icon should have a pointer cursor (in Burg overview)
* Lock tooltip is not very clear on what lock is doing. The tooltip can be more detailed (in both Burg overview and Burg Edit)

Legogizmo commented:
Ok, I'll fix it up and update.

Legogizmo commented:
Fixed the tool tips to be a bit more descriptive and Updated existing tool tips to mention they would only modify unlocked burgs.
Cultures Editor no longer renames burgs. 
Also managed to prevent burg group removal from removing locked burgs.

Azgaar commented:
Thanks! Reviewing and merging is everything is good :)
